src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -300,6 +304,162 @@ topSort g = runContT (evalStateT (topSort' g) initialState) id where
 isAcyclic :: AdjacencyIntMap -> Bool
 isAcyclic = isRight . topSort
 
+-- TODO: Benchmark and optimise.
```

Comment:
Is this TODO still relevant? As far as I can tell, you've done both!

Reply:
True!